lib/msf/ui/console/command_dispatcher/session.rb

```diff
@@ -0,0 +1,235 @@
+            if args.include?('-h') || args.include?('--help')
+              cmd_pry_help
+              return
+            end
+
+            begin
+              require 'pry'
+            rescue LoadError
+              print_error('Failed to load Pry, try "gem install pry"')
+              return
+            end
+
+            print_status('Starting Pry shell...')
+            print_status("You are in the \"client\" (session) object\n")
+
+            Pry.config.history_load = false
+            client.framework.history_manager.with_context(history_file: Msf::Config.pry_history, name: :pry) do
+              client.pry
+            end
+          end
+
+          def cmd_sessions_help
+            print_line('Usage: sessions <id>')
+            print_line
+            print_line('Interact with a different session Id.')
+            print_line('This works the same as calling this from the MSF shell: sessions -i <session id>')
+            print_line
+          end
+
+          def cmd_sessions(*args)
+            if args.empty? || args[0].to_i == 0
+              cmd_sessions_help
+            elsif args[0].to_s == client.name.to_s
+              print_status("Session #{client.name} is already interactive.")
+            else
+              print_status("Backgrounding session #{client.name}...")
+              # store the next session id so that it can be referenced as soon
+              # as this session is no longer interacting
+              client.next_session = args[0]
+              client.interacting = false
+            end
+          end
+
+          def cmd_resource_help
+            print_line 'Usage: resource path1 [path2 ...]'
+            print_line
+            print_line 'Run the commands stored in the supplied files. (- for stdin, press CTRL+D to end input from stdin)'
+            print_line 'Resource files may also contain ERB or Ruby code between <ruby></ruby> tags.'
+            print_line
+          end
+
+          def cmd_resource(*args)
+            if args.empty? || args.include?('-h') || args.include?('--help')
+              cmd_resource_help
+              return false
+            end
+
+            args.each do |res|
+              good_res = nil
+              if res == '-'
+                good_res = res
+              elsif ::File.exist?(res)
+                good_res = res
+              elsif [
+                ::Msf::Config.script_directory + ::File::SEPARATOR + 'resource' + ::File::SEPARATOR + 'meterpreter',
+                ::Msf::Config.user_script_directory + ::File::SEPARATOR + 'resource' + ::File::SEPARATOR + 'meterpreter'
+              ].each do |dir|
+                res_path = dir + ::File::SEPARATOR + res
+                if ::File.exist?(res_path)
+                  good_res = res_path
+                  break
+                end
+              end
+                # let's check to see if it's in the scripts/resource dir (like when tab completed)
+              end
```

Comment:
where's end's friend? he shares an indentation level with another end.

Reply:
huh, most of this was copied over from the meterpreter command_dispatcher, I swear I rubocopped this but there's definitely still issues, I'll get it sorted cheers